Pasting an emoji into the file finder panics

### Summary

Reproducible panic when pasting text with an emoji and a newline into file finder.

Steps to trigger the problem:
1. Copy this to your clipboard
```
Hi there! 👋
We're workin
```
2. Open file finder
3. Paste

Expected Behavior: No Panic

Actual Behavior: Panic

### Zed Version and System Specs

Zed: v0.179.0 (Zed Nightly 798a34bfc269f55f0b4c3fba6ade75a30a7c88a3) 
OS: macOS 15.3.1
Memory: 64 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

```
2025-03-14T09:53:51.67491-04:00 [ERROR] {
  "thread": "main",
  "payload": "byte index 11 is not a char boundary; it is inside '👋' (bytes 8..12) of `Hithere!👋\nWe'reworkin`",
  "location_data": {
    "file": "crates/util/src/paths.rs",
    "line": 318
  },
  "backtrace": [
    "<base64::engine::general_purpose::GeneralPurpose as base64::engine::Engine>::internal_decode::h83d604cec422db27+12217436",
    "<base64::engine::general_purpose::GeneralPurpose as base64::engine::Engine>::internal_decode::h83d604cec422db27+12217244",
    "text::Buffer::apply_local_edit::h1e7f6a5b77f8f3bf+88752080",
    "std::io::Write::write_fmt::hd4c273f34c4c0a33+70187848",
    "std::io::Write::write_all::h7713b80a44059b92+70186676",
    "std::io::stdio::Stdout::lock::hc0bca220eb18dab0+70181044",
    "std::io::default_read_to_end::hbc381ae94d02c57c+70185848",
    "std::sys::thread_local::native::lazy::Storage<T,D>::initialize::hf08d159c078cfe97+98845200",
    "core::num::bignum::Big32x40::mul_digits::h262a495938188cf7+18857152",
    "anyhow::error::<impl anyhow::Error>::construct::hd8059d5853df20ae+98846292",
    "<usvg::tree::Group as core::clone::Clone>::clone::hacec7021e8187cff+77182492",
    "file_finder::FileFinderDelegate::lookup_absolute_path::{{closure}}::{{closure}}::hb90c447f0aa2d9b6+31814704",
    "editor::selections_collection::MutableSelectionsCollection::select_ranges::h59ff116c00156694+31492388",
    "gpui::app::App::observe_new::{{closure}}::ha214c5d91df3231e+31212784",
    "gpui::app::App::new_subscription::h8149bb7ab00bc602+36160928",
    "gpui::window::Window::new::{{closure}}::{{closure}}::h04db14cf82968ad7+36749652",
    "gpui::platform::mac::window::selected_range::h0c96900c08559439+36603800",
    "gpui::platform::mac::window::MacWindowState::move_traffic_light::h50591fb28f40eba6+36566600",
    "<gpui::platform::mac::text_system::MacTextSystem as gpui::platform::PlatformTextSystem>::font_id::h511e22964109354e+36496520",
    "core::ptr::drop_in_place<search::buffer_search::BufferSearchBar>::h777c5f1c7d90fb27+87281720",
    "<D as sum_tree::SeekTarget<S,D>>::cmp::h61d89e96d4736452+88773084",
    "gpui::app::entity_map::WeakEntity<T>::update::hee0d8a3fe0ff2b38+88619972",
    "<core::str::error::Utf8Error as core::fmt::Debug>::fmt::h135368b3912581ab+89661180",
    "gimli::read::aranges::ArangeHeader<R,Offset>::parse::hf19a2e7a2068f89e+70123508",
    "sum_tree::SumTree<T>::from_child_trees::hc73410d2ccefa2a5+88784504"
  ],
  "app_version": "0.179.0",
  "app_commit_sha": "b5a7fb13c38838a3f0e4fd5fdc9b4c99df0c834e",
  "release_channel": "nightly",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "15.3.1",
  "architecture": "aarch64",
  "panicked_on": 1741960431674,
  "system_id": "e6b5b705-51bf-4704-80d7-4964627c0c04",
  "installation_id": "9146990d-7463-4795-bdf0-9357bd824f4b",
  "session_id": "54c14144-d653-4dfc-bf09-49f7abcba6f9"
}
```

</details>

--- a/crates/util/src/paths.rs
+++ b/crates/util/src/paths.rs
@@ -169,7 +169,7 @@ impl<T: AsRef<Path>> From<T> for SanitizedPath {
 /// A delimiter to use in `path_query:row_number:column_number` strings parsing.
 pub const FILE_ROW_COLUMN_DELIMITER: char = ':';
 
-const ROW_COL_CAPTURE_REGEX: &str = r"(?x)
+const ROW_COL_CAPTURE_REGEX: &str = r"(?xs)
     ([^\(]+)(?:
         \((\d+)[,:](\d+)\) # filename(row,column), filename(row:column)
         |
@@ -621,6 +621,24 @@ mod tests {
             PathWithPosition {
                 path: PathBuf::from("test_file.rs"),
                 row: Some(1),
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("ab\ncd"),
+            PathWithPosition {
+                path: PathBuf::from("ab\ncd"),
+                row: None,
+                column: None
+            }
+        );
+
+        assert_eq!(
+            PathWithPosition::parse_str("👋\nab"),
+            PathWithPosition {
+                path: PathBuf::from("👋\nab"),
+                row: None,
                 column: None
             }
         );

